### PR TITLE
Fixes #5882 - allow Foreman app to reach dynflow socket

### PR DESCRIPTION
**Pull request by lzap**

### CODE REVIEW

File: foreman.te [hunk 1]
```diff
@@ -213,6 +218,10 @@ optional_policy(`
         # Fedora.
         allow httpd_t puppet_etc_t:dir getattr;
         allow httpd_t puppet_port_t:tcp_socket name_connect;
+        # Puppet master does touch xattr which SELinux sees as relabelto
```
Comment (domcleal) [inline foreman.te:221]: This appears unrelated to dynflow...

File: foreman.te [hunk 2]
```diff
@@ -168,6 +168,11 @@ optional_policy(`
         # Allow Foreman to write to the SQlite databases
         read_files_pattern(passenger_t, foreman_db_t, foreman_db_t)
         write_files_pattern(passenger_t, foreman_db_t, foreman_db_t)
+
+        # Allow Foreman to reach Dynflow engine
+        allow passenger_t foreman_var_run_t:sock_file write;
+        init_stream_connect_script(passenger_t)
+        unconfined_stream_connect(passenger_t)
```
Comment (domcleal) [inline foreman.te:175]: That sounds fairly open?

**Comment by domcleal** [comment]
What AVCs are generated?  The bug report doesn't say.

File: foreman.te [hunk 3]
```diff
@@ -168,6 +168,11 @@ optional_policy(`
         # Allow Foreman to write to the SQlite databases
         read_files_pattern(passenger_t, foreman_db_t, foreman_db_t)
         write_files_pattern(passenger_t, foreman_db_t, foreman_db_t)
+
+        # Allow Foreman to reach Dynflow engine
```
Comment (domcleal) [inline foreman.te:172]: If this is a plugin, then it should probably be in that section.

File: foreman.te (hunk 1 above)
Comment (lzap) [inline foreman.te:221]: Yes, but I have the denials in the BZ. You you want me to split the upstream?

**Comment by lzap** [comment]
https://bugzilla.redhat.com/show_bug.cgi?id=1098244

File: foreman.te (hunk 1 above)
Comment (domcleal) [inline foreman.te:221]: Yes.

File: foreman.te (hunk 2 above)
Comment (lzap) [inline foreman.te:175]: Hmm yes I am out of ideas how to prevent Foreman from failing in enforcing when dynflow (unconfined) is connecting to Foreman via sockets.

**Comment by domcleal** [comment]
Ok, but please keep our redmine tickets up to date and relevant, they're what we use in Foreman, not bugzilla.  Use Bryan's clone script if you need to.

File: foreman.te (hunk 3 above)
Comment (lzap) [inline foreman.te:172]: Well I postponed delivering selinux policy for tasks because I spent whole day wrinting that and testing that and the result is not yet ready. I am not sure I can deliver that by end of friday. Daemons rubygem is the root cause of this.

File: foreman.te (hunk 2 above)
Comment (lzap) [inline foreman.te:175]: This is really pain, I need to write the policy for foreman-tasks really. This is not a solution, permissive is better than merging this.

**Comment by lzap** [comment]
Ok closing this one, I need to finish my proper solution to that.
